src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -225,25 +256,41 @@ private constructor(
             GraphQlLogger.logCodyEvent(project, "command:${commandId.displayName}", "executed")
           })
 
-      chatSession.addMessage(ChatMessage(Speaker.HUMAN, commandId.displayName))
+      chatSession.addMessageAtIndex(
+          ChatMessage(
+              Speaker.HUMAN,
+              source = commandId.source,
+              commandId.displayName,
+          ),
+          chatSession.messages.count())
+      chatSession.addMessageAtIndex(
```

Comment:
Can we force `shouldAddBlinkingCursor = true` there?
It will guarantee that cursor started blinking even if there is no response from the agent yet, it will be more responsive in some scenarios (e.g restart of the agent, slow LLM, bad connectivity)

Reply:
ok, but I tested it with no interent connection and it does not show the cursor (removes it immedietaly). 

```
cancellationToken.onFinished {
      ApplicationManager.getApplication().invokeLater {
        removeBlinkingCursor()
        getLastMessage()?.onPartFinished()
      }
    }
```

`cancellationToken` removes it when there is not internet